### PR TITLE
Code base added for internal review

### PR DESCRIPTION
**Pull request by bhushankawadkar**
Hydrator plugins code base added for internal review

### CODE REVIEW

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                                               " does not matched with file(s) name, inside path provided.");
+        }
+      }
+      //filter files from pre-processed file list.
+      listStatus = getPreProcessedFilteredListStatus(job.getConfiguration(), listStatus);
+    }
+    return listStatus;
+  }
+
+  private List<FileStatus> getPreProcessedFilteredListStatus(Configuration conf, List<FileStatus> listStatusList) {
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java]: Add description to method

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        if (!processedFileList.contains(filePath)) {
+          includedFileListStatus.add(listStatus);
+        }
+      }
+      return includedFileListStatus;
+    }
+    return listStatusList;
+  }
+
+  private List<String> getPreProcessedFileList(String processedDataTempFile) {
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java]: Add description for method

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    try {
+      //read temporary file containing processed file information.
+      File file = new File(processedDataTempFile);
+      BufferedReader reader = new BufferedReader(new FileReader(file));
+      String line = null;
+      while ((line = reader.readLine()) != null) {
+        processedFileList.add(line.trim());
+      }
+      reader.close();
+      //empty file contents
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java]: This comment can be removed

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      PrintWriter writer = new PrintWriter(file);
+      writer.print("");
+      writer.close();
+    } catch (IOException exception) {
+      //sallow exception
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java]: Provide appropriate error message in exception

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java [hunk 5]
```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright © 2015-2016 Cask Data, Inc.
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java]: CASK folks want this to only be 2016

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+
+  @VisibleForTesting
+  XMLReaderConfig getConfig() {
+    return config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    //validation of fields
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java]: can remove this comment

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+    //set file tracking information to be available to read.
+    setFileTrackingInfo(context, conf);
+
+    if (StringUtils.isNotEmpty(this.config.actionAfterProcess)) {
+      conf.set(XMLInputFormat.XML_INPUTFORMAT_FILE_ACTION, this.config.actionAfterProcess);
+    }
+    if (StringUtils.isNotEmpty(this.config.actionAfterProcess)) {
+      conf.set(XMLInputFormat.XML_INPUTFORMAT_TARGET_FOLDER, this.config.targetFolder);
+    }
+
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java]: Extra line. Remove it

Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java]: There are few extra lines. See if they can be removed

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+      } else if (Strings.isNullOrEmpty(this.nodePath)) {
+        throw new IllegalArgumentException("Node path cannot be empty.");
+      } else if (!Strings.isNullOrEmpty(this.actionAfterProcess) && this.reprocessingReq.equalsIgnoreCase("YES")) {
+        throw new IllegalArgumentException("Please select either 'After Processing Action' or 'Reprocessing " +
+                                             "Required', both cannot be applied at same time.");
+      } else if (this.actionAfterProcess != null && (this.actionAfterProcess.equalsIgnoreCase("archive") ||
+        this.actionAfterProcess.equalsIgnoreCase("move")) && Strings.isNullOrEmpty(this.targetFolder)) {
+        throw new IllegalArgumentException("Target folder cannot be Empty for Action = " + actionAfterProcess + ".");
+      }
+      else if (Strings.isNullOrEmpty(this.tableName)) {
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java]: Should this be after nodePath check as this is check for mandatory fields which should happen with other mandatory fields?

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLRecordReader.java [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+  }
+
+  private void actionsAfterEOF() throws IOException {
+    if (StringUtils.isNotEmpty(fileAction)) {
+      processFileActions();
+    }
+    updateFileTrackingInfo();
+  }
+
+  /**
```
Comment (ameyk) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLRecordReader.java]: Add description to method

**Comment by ameyk** [comment]
Most of the comments discussed offline for XML Reader were already incorporated. I have given few more comments.

**Comment by ameyk** [comment]
LGTM

File: hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java (hunk 3 above)
Comment (bhushankawadkar) [inline hydrator-plugins/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java]: This comment removed bu added description for why making it empty.

**Pull request merged**